flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/connectors/DynamicSourceUtils.java

```diff
@@ -71,6 +71,9 @@
 @Internal
 public final class DynamicSourceUtils {
 
+    // Ensures that physical and metadata columns don't collide.
+    public static final String METADATA_COLUMN_PREFIX = "$metadata$";
```

Comment:
minor, but maybe we can define this constant in a common place for both source and sink?

Reply:
the source and sink stack are completely separate, that's why we also have separate util classes